assets/javascripts/kitten/components/form/autocomplete/index.js

```diff
@@ -0,0 +1,293 @@
+  }
+`
+
+export const Autocomplete = ({
+  items: defaultItems,
+  error,
+  onChange,
+  onBlur,
+  onKeyDown,
+  ...props
+}) => {
+  const [items, setItems] = useState(defaultItems)
+  const [value, setValue] = useState(props.defaultValue)
+  const [selectedItemIndex, setSelectedItemIndex] = useState(-1)
+  const [showSuggestions, setShowSuggestions] = useState(false)
+  const inputEl = useRef(null)
+  const suggestionsEl = useRef(null)
+
+  useEffect(() => {
+    updateSuggestions()
+    setShowSuggestions(!!value)
+  }, [value])
+
+  const handleChange = e => {
+    setValue(e.target.value)
+
+    onChange()
+  }
+
+  const handleBlur = () => {
+    setTimeout(() => {
+      setShowSuggestions(false)
+    }, 100)
+
+    onBlur()
+  }
+
+  const handleKeyDown = e => {
+    const enterKeyCode = 13
+    const escKeyCode = 27
+    const arrowUpKeyCode = 38
+    const arrowDownKeyCode = 40
+
+    if (e.keyCode === escKeyCode) setShowSuggestions(false)
+
+    if (showSuggestions) {
+      if (e.keyCode === arrowUpKeyCode) {
+        e.preventDefault()
+        prevSelectedItem()
+      }
+
+      if (e.keyCode === arrowDownKeyCode) {
+        e.preventDefault()
+        nextSelectedItem()
+      }
+
+      if (e.keyCode === enterKeyCode) {
+        e.preventDefault()
+
+        const selectedValue = items[selectedItemIndex]
+        handleClickItem(selectedValue)()
+      }
+    }
+
+    onKeyDown()
+  }
+
+  const handleClickItem = value => () => {
+    if (!value) return
+
+    inputEl.current.value = value
+    inputEl.current.focus()
+
+    setValue(value)
+    setShowSuggestions(false)
+  }
+
+  const updateSuggestions = () => {
+    const regex = new RegExp(`^${value}`, 'gi')
+    const newItems = defaultItems.filter(
+      item => item.match(regex) && item !== value,
+    )
+
+    setItems(newItems)
+    resetSelectedItem()
+  }
+
+  const prevSelectedItem = () => {
+    const newIndex = selectedItemIndex - 1
+
+    suggestionsScrollTop(newIndex)
+    setSelectedItemIndex(newIndex < 0 ? 0 : newIndex)
+
+    const scrollTop = newIndex * itemHeight
+    suggestionsScrollTop(scrollTop)
+  }
+
+  const nextSelectedItem = () => {
+    const newIndex = selectedItemIndex + 1
+    const itemsLength = items.length - 1
+
+    suggestionsScrollTop(newIndex)
+    setSelectedItemIndex(newIndex >= itemsLength ? itemsLength : newIndex)
+
+    const scrollTop = (newIndex - maxItemsVisibled + 1) * itemHeight
+    suggestionsScrollTop(scrollTop)
+  }
+
+  const resetSelectedItem = () => setSelectedItemIndex(-1)
+
+  const suggestionsScrollTop = value => {
+    suggestionsEl.current.scrollTop = value > 0 ? value : 0
+  }
+
+  return (
+    <Container>
+      <Input
+        {...props}
+        error={error}
+        ref={inputEl}
+        type="text"
+        autoComplete="false"
+        onChange={handleChange}
+        onBlur={handleBlur}
+        onKeyDown={handleKeyDown}
+        aria-owns={`${props.name}-results`}
```

Comment:
Trop bien ! Je ne connaissais pas cet attribut !

Reply:
C'est un attribut super utile pour les relations parent/enfant : ça rajoute une relation dans l'arbre d'accessibilité sans avoir à changer le DOM https://tink.uk/using-the-aria-owns-attribute/